src/lib/coerce.js

```diff
@@ -309,3 +335,22 @@ exports.coerceFont = function(coerce, attr, dfltObj) {
 
     return out;
 };
+
+exports.validate = function(value, opts) {
+    var valObject = exports.valObjects[opts.valType];
+
+    if(opts.arrayOk && Array.isArray(value)) return true;
+
+    if(valObject.validateFunction) {
+        return valObject.validateFunction(value, opts);
+    }
+
+    var failed = {},
+        out = failed,
+        propMock = { set: function(v) { out = v; } };
+
+    // 'failed' just something mutable that won't be === anything else
+
+    valObject.coerceFunction(value, propMock, failed, opts);
```

Comment:
so the
```
failed = {}
out = failed
out = v
coerce()
out !== failed
```
dance is just checking to see if if propMock is fired by checking to see if the exising `out === failed` reference is no longer valid?